### ISSUE TITLE
[Bug]: Apps that follow the OS's dark/light mode always detect dark

### ISSUE THREAD

**Issue by hoodust** [prasanthrangan/hyprdots#797]
### Description

Applications like Brave browser, Discord, Bluemail, etc. that have a setting to mirror their dark/light modes to match the OS (i.e. "Sync with OS", "Use Linux Settings", etc.) never change and always think the OS is dark.

### Steps to Reproduce

1. Install any app (such as Discord) that has a setting to follow the OS theme.
2. Enable that setting in-app (in case of Discord: User Settings > Appearance > Theme > Sync with Computer)
3. Set a light theme, such as Catppuccin Latte or Frosted Glass.
4. Observe no change in the application, with it remaining dark, even after restarting it.

### Expected Behavior

App would switch to light/dark mode based on theme.

### Actual Behavior

App stays dark at all times, requiring you to manually set light/dark mode every theme change.

### Screenshots

![image](https://github.com/prasanthrangan/hyprdots/assets/69404732/a896cc57-9c78-45ea-b2ee-707866ff95d9)

### Environment

- Arch
- Kernel: 6.6.10-arch1-1
- Hyprland Version: v0.32.3-74-gad3f6886
- Installation Environment: desktop/bare-metal/non-vm

Thanks!!

**Comment by kRHYME7** [comment]
flatpak?

**Comment by hoodust** [comment]
> flatpak

No, `yay -S`. Never used flatpak, should I?

**Comment by kRHYME7** [comment]
On this theme, Do Brave won't shift to light mode too?

**Comment by hoodust** [comment]
Interesting. Nope, for me when I switch to Catppuccin-Latte, Frosted Glass, or Material Sakura, Brave (and discord and bluemail) all stay dark, even if I close and restart them. In Brave settings I have Appearance > Brave Colors set to "Same as Linux".

**Comment by kRHYME7** [comment]
I see, I'm not home right now. But I think you have some GTK Problem?  Try to use nwg-look then try to toogle the color scheme

**Comment by hoodust** [comment]
Ah, so nwg-look got set to "Default" whatever that means, but changing the color scheme there worked (sort of) and after closing it, the theme switcher now works too (sort of) so maybe the "Default" setting was the problem. 

It's far from perfect, needing to completely close Discord and Bluemail (including the tray icon, not just the window) but they're switched when re-opened (kinda like Dolphin). Brave actually switches now, and without having to close it, so that's nice!

So is that as good as it gets: some apps (Dolphin, Discord, Bluemail, etc.) need to be closed and reopened for the light/dark mode to "take"?

**Comment by kRHYME7** [comment]
nope, try to run the themeswitch script to see some outputs maybe we have permission errors

**Comment by hoodust** [comment]
```
❯ ~/.config/hypr/scripts/themeselect.sh
Selected theme: Tokyo-Night
DP-2: 3440x1440, scale: 1, currently displaying: image: /home/hoo/.config/swww/Catppuccin-Latte/waifu_jasmin.jpg
/home/hoo/.config/hypr/scripts/themeswitch.sh: line 112: flatpak: command not found
/home/hoo/.config/hypr/scripts/themeswitch.sh: line 113: flatpak: command not found
ok
environment: line 5: /home/hoo/.config/spicetify/Themes/Sleek/Wall-Dcol.ini: No such file or directory
sed: can't read /home/hoo/.config/spicetify/Themes/Sleek/Wall-Dcol.ini: No such file or directory
```

**Comment by hoodust** [comment]
flatpak errors make sense as it's not even installed, and I don't use spotify/spicetify (if hyprdots installed it then I probably removed it). Other than those it seems fine. 

nwg-look throws a bunch of "Theme parsing error" messages in terminal, but idk if that's a problem.

I tested on a laptop I previously demoed hyprdots on before I moved to it on my desktop, and the same apps don't change without restarting them on there also, but you're saying Dolphin, Discord, etc. change instantly for you?

**Comment by kRHYME7** [comment]
> but you're saying Dolphin, Discord, etc. change instantly for you?

Yes, it should change instantly. 


Can you just remove the flatpak-related lines or add a condition for future use case?

```
if which flatpak > /dev/null; then
    flatpak --user override --env=GTK_THEME="${ThemeSet}" || true
    flatpak --user override --env=ICON_THEME="${IconSet}" || true
fi
```



You got a bunch of errors here Maybe the script did not complete. 
Try to disable wallbash first so we can run it cleanly

**Comment by hoodust** [comment]
Interesting, it's never worked for me on my laptop or desktop since I started using hyprdots in mid-November. It's just that I've started switching themes more lately and it's been bugging me. In both cases I followed the readme (minimal Arch install + clone and run hyprdots).

> Try to disable wallbash first so we can run it cleanly

Sorry, how do I do that? You mean just comment out the last line in themeswitch.sh?

**Comment by hoodust** [comment]
Oh and not sure why but after a reboot Brave no longer switches automatically again (stays dark even restarting it).

**Comment by hoodust** [comment]
I discovered that some of the scripts don't match the latest in github (such as themeselect.sh, themeswitch.sh, and globalcontrol.sh). In particular line 27 of themeselect.sh does not include `-select "${gtkTheme}"` which seemed like a promising lead.

So on a test installation on another drive I did a git pull and ran `install.sh -r` with the default restore_cfg.lst but it didn't fix anything.

I'm tempted to do another fresh install of Arch and Hyprdots on that other drive just to see if it the theming works like you said it should, because it never has for me for some reason.

**Comment by kRHYME7** [comment]
did you git pull to the latest changes?  No need for a fresh install thu.  Just run ``` install.sh ``` to overwrite

**Comment by hoodust** [comment]
`Just run install.sh to overwrite`
Just tried this on test install, even with changing everything to Y in restore_cfg.lst it made no change to the theme problem.

`No maybe just toggle the wallbashtoggle script`
Toggled off (ugly swaybar) and re-ran themeselect.sh. No change to the theme problem. Although strangely I get these new errors now:
```
rm: cannot remove '/home/hoo/.config/gtk-4.0': Is a directory
ln: failed to create symbolic link '/home/hoo/.config/gtk-4.0/gtk-4.0': File exists
```

**Comment by Enigma1309** [comment]
> flatpak?

i have the similiar problem my flatpak apps don't change theme. what should i do?

**Comment by kRHYME7** [comment]
``` gsettings set org.gnome.desktop.interface color-scheme 'prefer-dark'    ``` Do this command affect the flatpak app? 

Or I remembered giving the flatpak some permission to the GTK directory

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/d56b13c8-22d6-44da-b7c9-52c0e0de841f)
I am not really sure but maybe that helps

**Comment by Enigma1309** [comment]
Thanks it worked

All i DID is 

1) `sudo cp -r /usr/share/themes ~/.themes`
2) Add this config in foliate global settings
![image](https://github.com/prasanthrangan/hyprdots/assets/94917588/8282414b-f009-487f-b650-1078e0b7eac8)

**Comment by kRHYME7** [comment]
Soon! I think we can solve this granted that Hyprdots will comply XDG

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/58970738-b4a3-4f87-85df-bf56c3da0cd4)

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/e6b0c7ff-50ca-4b3c-8c49-1538dcf94c5c)

This way flatpak might be able to identify that

**Comment by kRHYME7** [comment]
@hoodust  Have you tried ``` gsettings set org.gnome.desktop.interface color-scheme 'prefer-dark'  ``` change prefer-light just to see if it toggles.

**Comment by hoodust** [comment]
@kRHYME7 Ah, I see. That makes sense. I opened this because initially even after _restarting_ these apps they didn't change, but I think this fixed it:

> Ah, so nwg-look got set to "Default" whatever that means, but changing the color scheme there worked (sort of) and after closing it, the theme switcher now works too (sort of) so maybe the "Default" setting was the problem.

...and it seems like the rest is expected behavior (having to restart non-gtk apps) which isn't a huge deal. I just got confused when you told me those should switch without restarting them.

Thanks for all your help!

**Comment by kRHYME7** [comment]
@hoodust  dolphin now follows configuration color and changes  it dynamically. Try the latest commit.
